diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/ExtensionFunctionsSameNameRule.kt

```diff
@@ -0,0 +1,140 @@
+
+/**
+ * This rule checks if extension functions with the same signature don't have related classes
+ */
+class ExtensionFunctionsSameNameRule(private val configRules: List<RulesConfig>) : Rule("extension-functions-same-name") {
+    private lateinit var emitWarn: ((offset: Int, errorMessage: String, canBeAutoCorrected: Boolean) -> Unit)
+    private var isFixMode: Boolean = false
+
+    override fun visit(node: ASTNode,
+                       autoCorrect: Boolean,
+                       emit: (offset: Int, errorMessage: String, canBeAutoCorrected: Boolean) -> Unit) {
+        emitWarn = emit
+        isFixMode = autoCorrect
+
+        /**
+         * 1) Collect all classes that extend other classes (collect related classes)
+         * 2) Collect all extension functions with same signature
+         * 3) Check if classes of functions are related
+         */
+        if (node.elementType == FILE) {
+            val relatedClasses = collectAllRelatedClasses(node)
+            val extFunctionsWithSameName = collectAllExtensionFunctions(node)
+            handleFunctions(relatedClasses, extFunctionsWithSameName)
+        }
+
+    }
+
+    // Fixme: should find all related classes in project, not only in file
+    @Suppress("UnsafeCallOnNullableType")
+    private fun collectAllRelatedClasses(node: ASTNode) : List<Pair<String, String>> {
+        val classListWithInheritance = node.findAllNodesWithSpecificType(CLASS)
+                .filterNot { (it.psi as KtClass).isInterface() }
+                .filter { it.hasChildOfType(SUPER_TYPE_LIST) }
+
+        val pairs = mutableListOf<Pair<String, String>>()
+        classListWithInheritance.forEach {
+            val callEntries = it.findChildByType(SUPER_TYPE_LIST)!!.getAllChildrenWithType(SUPER_TYPE_CALL_ENTRY)
+
+            callEntries.forEach { entry ->
+                if (entry.hasChildOfType(VALUE_ARGUMENT_LIST)) {
+                    val className = (it.psi as KtClass).name!!
+                    val entryName = entry.findLeafWithSpecificType(IDENTIFIER)!!
+                    pairs.add(Pair(className, entryName.text))
+                }
+            }
+        }
+        return pairs
+    }
+
+    @Suppress("UnsafeCallOnNullableType")
+    private fun collectAllExtensionFunctions(node: ASTNode) : SimilarSignatures {
+        val extensionFunctionList = node.findAllNodesWithSpecificType(FUN).filter { it.hasChildOfType(TYPE_REFERENCE) && it.hasChildOfType(DOT) }
+        val distinctFunctionSignatures = mutableMapOf<FunctionSignature, ASTNode>() // maps function signatures on node it is used by
+        val extensionFunctionsPairs = mutableListOf<Pair<ExtensionFunction, ExtensionFunction>>() // pairs extension functions with same signature
+
+        extensionFunctionList.forEach {
+            val functionName = (it.psi as KtNamedFunction).name!!
+            val params = it.getFirstChildWithType(VALUE_PARAMETER_LIST)!!.text
```

Comment:
Comparing by text only can lead to errors: there may be additional spaces or line breaks or comments in one function and not in the other. You should compare parameters themselves.